### DISCUSSION TITLE
Specifying additional docker network for nextcloud-aio-apache container

### DISCUSSION THREAD

**Discussion by apparle** [nextcloud/all-in-one#5482] [Ideas] [answered]
When setting up a reverse proxy (like Nginx Proxy Manager) in Docker container, not on the host, one of the most recommended patterns is to create a separate network for frontend, and then separate isolated backend networks for each of the services. 
But because AIO creates its own network and bundles everything under it, this becomes difficult. Further because AIO setup mandates the name be `nextcloud-aio`, the only option is to have the reverse proxy container join nextcloud-aio breaking the desired isolation across services. 

Since nextcloud already has the Apache container as the front facing entity, it'll be cleaner to connect just the nextcloud-aio-apache container to the frontend network, and keep all the nextcloud AIO maintained containers into the internal `nextcloud-aio`.

To that end, there should be an option to specify an additional external network that apache container should connect to. 
I've implemented this in the form of an evironment variable `APACHE_CONTAINER_ADDITIONAL_NETWORK` that can be configured while starting the mastercontainer. There was a similar discussion in past https://github.com/nextcloud/all-in-one/discussions/4714 but it only stopped at the workaround above.

Here's the first cut implementation which I've tested locally:
https://github.com/nextcloud/all-in-one/compare/main...apparle:apache_additional_network?expand=1 

I can generate the PR but first wanted to open the discussion on whether this is the right way, since I'm very new to php. There will be some documentation update needed, which I can add when creating the PR.

**Comment by szaimen** [comment, marked as answer]
Hi, thanks for reaching out regarding this. 

My suggestion would be that you create the PR and we do further discussion in there. 

Best would be if you could already provide the update to the documentation directly within the PR.

Thanks a lot! :)

**Reply by szaimen** [reply]
This is getting worked on in https://github.com/nextcloud/all-in-one/pull/5484

**Reply by szaimen** [reply]
This is now released with v9.9.0 Beta. Testing and feedback is welcome! See https://github.com/nextcloud/all-in-one#how-to-switch-the-channel

**Reply by WaitingAtTheCrossroads** [reply]
Excellent timing.  I was just setting up a new install and needing just such a feature.  I switched to 9.9.0 beta and it works as advertised. Thanks! :smile:
